echo_server.py

```diff
@@ -50,22 +52,45 @@ def response_error(error_num, error_msg):
 
 def parse_request(request):
     request_list = request.split()
-    print len(request_list)
-    if len(request_list) != 5:
-        return None
     error_check = check_for_errors(request_list)
     if error_check == "Good to go!":
-        return response_ok(request_list[1])
+        body_and_type = resolve_uri(request_list[1])
+        return response_ok(request_list[1], body_and_type)
```

Comment:
We talked in class today about why nesting functions this way is problematic.  Your tests are a symptom of the issue.